src/datepicker/calendar.js

```diff
@@ -397,11 +397,7 @@ export default class Calendar<T = Date> extends React.Component<
     const {value} = this.props;
     const selected = this.getSingleDate(value);
     let nextState;
-    if (
-      selected &&
```

Comment:
I think the fix here would be to check if it's a start or end date. See the code change suggestions.

Reply:
If this change is made, then for case of monthsShown={3} when the mouseleave happens outside range, the date wont be highlighted, is that okay?
<img width="1075" alt="Screenshot 2020-08-08 at 11 21 38 AM" src="https://user-images.githubusercontent.com/26969693/89703539-60f9d280-d969-11ea-9e5b-d5977b91884d.png">